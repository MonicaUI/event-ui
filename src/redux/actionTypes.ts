export const LOAD_ACCOUNT_SNAPSHOT = 'event-guestside-site/LOAD_ACCOUNT_SNAPSHOT';
export const LOAD_EVENT_SNAPSHOT = 'event-guestside-site/LOAD_EVENT_SNAPSHOT';
export const LOAD_WEBSITE_CONTENT = 'event-guestside-site/LOAD_WEBSITE_CONTENT';
export const SET_REFERRER = 'event-guestside-site/SET_REFERRER';
export const LOAD_REGISTRATION_PROCESS_CONTENT = 'event-guestside-site/LOAD_REGISTRATION_PROCESS_SNAPSHOT';
export const LOAD_VISIBLE_PRODUCTS = 'event-guestside-site/LOAD_VISIBLE_PRODUCTS';
export const LOAD_REG_CART_VISIBLE_PRODUCTS = 'event-guestside-site/LOAD_REG_CART_VISIBLE_PRODUCTS';
export const REPLACE_VISIBLE_SESSION_PRODUCTS = 'event-guestside-site/REPLACE_VISIBLE_SESSION_PRODUCTS';
export const LOAD_VISIBLE_SESSION_PRODUCTS = 'event-guestside-site/LOAD_VISIBLE_SESSION_PRODUCTS';
export const LOAD_VISIBLE_PRODUCTS_BY_REG_TYPE = 'event-guestside-site/LOAD_VISIBLE_PRODUCTS_BY_REG_TYPE';
export const LOAD_EVENT_WEB_APP_SETTINGS = 'event-guestside-site/LOAD_EVENT_WEB_APP_SETTINGS';
export const SET_WEBSITE_PASSWORD_VALIDITY = 'event-guestside-site/SET_WEBSITE_PASSWORD_VALIDITY';
export const UPDATE_QUESTION_VISIBILTY = 'event-guestside-site/UPDATE_QUESTION_VISIBILITY';
export const GET_ORDERS = 'event-guestside-site/GET_ORDERS';
export const CLEAR_ORDERS = 'event-guestside-site/CLEAR_ORDERS';
export const SWITCH_CCPA_DIALOG_MODE = 'event-guestside-site/SWITCH_CCPA_DIALOG_MODE';
export const SHOW_TRANSPARENT_WRAPPER = 'event-guestside-site/SHOW_TRANSPARENT_WRAPPER';
export const HIDE_TRANSPARENT_WRAPPER = 'event-guestside-site/HIDE_TRANSPARENT_WRAPPER';
export const SPINNER_SELECTION_PENDING = 'event-guestside-site/SPINNER_SELECTION_PENDING';
export const SPINNER_SELECTION_DONE = 'event-guestside-site/SPINNER_SELECTION_DONE';
export const RESTORE_ALL_REG_TYPES_FOR_EVENT = 'event-guestside-site/RESTORE_ALL_REG_TYPES_FOR_EVENT';
