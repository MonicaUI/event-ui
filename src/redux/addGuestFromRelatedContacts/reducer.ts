const TOGGLE_RELATED_CONTACTS_VIEW = 'event-guestside-site/addGuestFromRelatedContacts/LIST_RELATED_CONTACTS';
const LOAD_RELATED_CONTACTS_FOR_INVITEE =
  'event-guestside-site/addGuestFromRelatedContacts/LOAD_RELATED_CONTACTS_FOR_INVITEE';
const LOAD_RELATED_CONTACTS_SEARCH_RESULTS =
  'event-guestside-site/addGuestFromRelatedContacts/LOAD_RELATED_CONTACTS_SEARCH_RESULTS';

export function setRelatedContacts(contactId: $TSFixMe, relatedContacts: $TSFixMe): $TSFixMe {
  return { type: LOAD_RELATED_CONTACTS_FOR_INVITEE, payload: { contactId, relatedContacts } };
}

export function toggleRelatedContacts(): $TSFixMe {
  return { type: TOGGLE_RELATED_CONTACTS_VIEW };
}

export function setRelatedContactsSearch(
  contactId: $TSFixMe,
  searchCriteria: $TSFixMe,
  searchResults: $TSFixMe
): $TSFixMe {
  return { type: LOAD_RELATED_CONTACTS_SEARCH_RESULTS, payload: { contactId, searchCriteria, searchResults } };
}

export default function addGuestFromRelatedContactsReducer(state = {}, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case LOAD_RELATED_CONTACTS_FOR_INVITEE: {
      return {
        ...state,
        [action.payload.contactId]: {
          ...state[action.payload.contactId],
          relatedContacts: action.payload.relatedContacts
        }
      };
    }
    case TOGGLE_RELATED_CONTACTS_VIEW: {
      return {
        ...state,
        showRelatedContactsView: !(state as $TSFixMe).showRelatedContactsView
      };
    }
    case LOAD_RELATED_CONTACTS_SEARCH_RESULTS: {
      return {
        ...state,
        [action.payload.contactId]: {
          ...state[action.payload.contactId],
          relatedContactsSearchCriteria: action.payload.searchCriteria,
          relatedContactsSearchResults: action.payload.searchResults
        }
      };
    }
    default: {
      return state;
    }
  }
}
