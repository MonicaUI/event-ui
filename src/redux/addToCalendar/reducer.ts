import { CALENDAR_EVENT_ID } from '../../utils/addToCalendarUtils';

export const LOAD_ADD_TO_CALENDAR_PROVIDERS = 'event-guestside-site/calendar/LOAD_CALENDAR_PROVIDERS';
export const LOAD_EVENT_CALENDAR = 'event-guestside-site/calendar/LOAD_EVENT_CALENDAR';
export const LOAD_SESSION_CALENDAR = 'event-guestside-site/calendar/LOAD_SESSION_CALENDAR';

export function setAddToCalendarProviders(calendarProviders: $TSFixMe): $TSFixMe {
  return {
    type: LOAD_ADD_TO_CALENDAR_PROVIDERS,
    payload: { calendarProviders }
  };
}

export function setCalendarUrl(calendarType: $TSFixMe, calendarUrl: $TSFixMe, entityId = CALENDAR_EVENT_ID): $TSFixMe {
  if (entityId === CALENDAR_EVENT_ID) {
    return {
      type: LOAD_EVENT_CALENDAR,
      payload: { calendarType, calendarUrl }
    };
  }

  return {
    type: LOAD_SESSION_CALENDAR,
    payload: { calendarType, calendarUrl, entityId }
  };
}

export default function addToCalendarReducer(state = {}, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case LOAD_ADD_TO_CALENDAR_PROVIDERS: {
      return {
        ...state,
        ...action.payload.calendarProviders
      };
    }
    case LOAD_EVENT_CALENDAR: {
      return {
        ...state,
        [action.payload.calendarType]: {
          ...state[action.payload.calendarType],
          [CALENDAR_EVENT_ID]: action.payload.calendarUrl
        }
      };
    }
    case LOAD_SESSION_CALENDAR: {
      return {
        ...state,
        [action.payload.calendarType]: {
          ...state[action.payload.calendarType],
          [action.payload.entityId]: action.payload.calendarUrl
        }
      };
    }
    default: {
      return state;
    }
  }
}
