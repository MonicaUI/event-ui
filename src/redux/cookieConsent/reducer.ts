// Reducer to update state for cookie Consent
import { DENY } from '../../utils/CookieConstants';

export const SET_COOKIE_CONSENT = 'event-guestside-site/cookieConsent/SET_COOKIE_CONSENT';

export default function cookieConsentReducer(state = { status: DENY }, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SET_COOKIE_CONSENT:
      return {
        ...state,
        status: action.payload.cookieStatus
      };
    default:
      return state;
  }
}
