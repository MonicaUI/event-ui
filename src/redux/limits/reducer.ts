const SET_ACCOUNT_LIMITS = 'event-guestside-site/limits/SET_ACCOUNT_LIMITS';

export const setAccountLimits = (accountLimits: $TSFixMe): $TSFixMe => {
  return {
    type: SET_ACCOUNT_LIMITS,
    payload: {
      accountLimits
    }
  };
};

export default function reducer(state: $TSFixMe, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SET_ACCOUNT_LIMITS: {
      return action.payload.accountLimits;
    }
    default: {
      return state;
    }
  }
}
