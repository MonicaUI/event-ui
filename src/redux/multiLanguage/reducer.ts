import { SET_LOCALE } from 'nucleus-widgets/redux/modules/text';
import { LOAD_LANGUAGE } from './actions';

export default function reducer(state: $TSFixMe, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SET_LOCALE:
      return {
        ...state,
        locale: action.payload.locale
      };
    case LOAD_LANGUAGE:
      return {
        ...state,
        loadedLanguages: [...state.loadedLanguages, action.payload]
      };
    default:
      return state;
  }
}
