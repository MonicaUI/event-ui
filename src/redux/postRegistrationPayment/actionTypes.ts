/* ACTION TYPES */

export const SUBMIT_WEB_PAYMENT = 'event-guestside-site/postRegistrationPayment/SUBMIT_WEB_PAYMENT';
export const RESET_SUBMIT_WEB_PAYMENT = 'event-guestside-site/postRegistrationPayment/RESET_SUBMIT_WEB_PAYMENT';
export const STORE_WEBPAYMENT_DATA = 'event-guestside-site/postRegistrationPayment/STORE_WEBPAYMENT_DATA';
export const FINALIZE_POSTREG_CHECKOUT_PENDING =
  'event-guestside-site/postRegistrationPayment/FINALIZE_POSTREG_CHECKOUT_PENDING';
export const FINALIZE_POSTREG_CHECKOUT_COMPLETE =
  'event-guestside-site/postRegistrationPayment/FINALIZE_POSTREG_CHECKOUT_COMPLETE';
