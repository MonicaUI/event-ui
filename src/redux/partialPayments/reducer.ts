import { UPDATE_PAYMENT_AMOUNT, UPDATE_SELECTED_PAYMENT_OPTION } from './actions';
import { PAYMENT_AMOUNT_OPTION } from 'event-widgets/utils/paymentConstant';
import { START_MODIFICATION_SUCCESS } from '../registrationForm/regCart/actionTypes';
import { GET_ORDERS } from '../actionTypes';

// Reducer to update state for Partial Payments
export default function reducer(state: $TSFixMe, action: $TSFixMe): $TSFixMe {
  const defaultPartialPayment = { paymentAmountOption: null, paymentAmount: null };
  switch (action.type) {
    case UPDATE_SELECTED_PAYMENT_OPTION: {
      const FULL_PAYMENT_VALUE = PAYMENT_AMOUNT_OPTION.FULL_PAYMENT.value;
      const paymentOption =
        action.payload === FULL_PAYMENT_VALUE
          ? PAYMENT_AMOUNT_OPTION.FULL_PAYMENT
          : PAYMENT_AMOUNT_OPTION.PARTIAL_PAYMENT;
      return {
        ...state,
        paymentAmountOption: paymentOption
      };
    }
    case UPDATE_PAYMENT_AMOUNT:
      return {
        ...state,
        paymentAmount: action.payload
      };
    case START_MODIFICATION_SUCCESS:
    case GET_ORDERS:
      return defaultPartialPayment;
    default:
      return state;
  }
}
