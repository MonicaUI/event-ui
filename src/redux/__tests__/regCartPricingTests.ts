import reducer from '../regCartPricing';
import { LOG_OUT_REGISTRANT_SUCCESS } from '../registrantLogin/actionTypes';

const initialRegCartPricing = reducer(undefined, {});

// Note that this is a shorted regCart for brevity.
const regCartPricing = {
  regCartId: '928f02df-4ba7-4637-824d-ae71bd414e4f',
  productFeeAmountCharge: 0,
  productFeeAmountRefund: 0,
  productSubTotalAmountCharge: 0,
  productSubTotalAmountRefund: 0,
  netFeeAmountCharge: 0,
  netFeeAmountRefund: 0
};

const logoutAction = { type: LOG_OUT_REGISTRANT_SUCCESS };

test('Verifying initial state.', () => {
  expect(reducer(undefined, {})).toMatchSnapshot();
});

test('LOG_OUT_REGISTRANT_SUCCESS does not change default pricing', () => {
  expect(reducer(initialRegCartPricing, logoutAction)).toMatchSnapshot();
});

test('LOG_OUT_REGISTRANT_SUCCESS sets pricing to default.', () => {
  // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ regCartId: string; productFeeA... Remove this comment to see the full error message
  expect(reducer(regCartPricing, logoutAction)).toMatchSnapshot();
});
