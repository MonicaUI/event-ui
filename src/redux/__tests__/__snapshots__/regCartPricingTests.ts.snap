// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LOG_OUT_REGISTRANT_SUCCESS does not change default pricing 1`] = `null`;

exports[`LOG_OUT_REGISTRANT_SUCCESS sets pricing to default. 1`] = `null`;

exports[`Verifying initial state. 1`] = `
Object {
  "isEditPrice": false,
  "isEditRefund": false,
  "plannerOverriddenProductFees": Object {},
  "plannerOverriddenProductRefunds": Object {},
}
`;
